src/error/MatterError.ts

```
@@ -0,0 +1,146 @@
+StatusResponseError.registerErrorClass(StatusCode.Timeout, TimeoutStatusResponseError);
+
+export class UnsupportedModeStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedMode, UnsupportedModeStatusResponseError);
+
+export class BusyStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.Busy, BusyStatusResponseError);
+
+export class UnsupportedClusterStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedCluster, UnsupportedClusterStatusResponseError);
+
+export class NoUpstreamSubscriptionStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.NoUpstreamSubscription, NoUpstreamSubscriptionStatusResponseError);
+
+export class NeedsTimedInteractionStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.NeedsTimedInteraction, NeedsTimedInteractionStatusResponseError);
+
+export class UnsupportedEventStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.UnsupportedEvent, UnsupportedEventStatusResponseError);
+
+export class PathsExhaustedStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.PathsExhausted, PathsExhaustedStatusResponseError);
+
+export class TimedRequestMismatchStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.TimedRequestMismatch, TimedRequestMismatchStatusResponseError);
+
+export class FailsafeRequiredStatusResponseError extends StatusResponseError {}
+StatusResponseError.registerErrorClass(StatusCode.FailsafeRequired, FailsafeRequiredStatusResponseError);
+
+
+export function isMatterError(e: unknown): e is MatterError {
+    return e instanceof MatterError;
+}
+
+type MatterErrorHandler<T> = (error: MatterError) => T | undefined;
```

This should not return undefined

---

In the concept here a return of undefined by the method would throw the error instead of using it as fallback value, so by intent

---

I also added JSDoc